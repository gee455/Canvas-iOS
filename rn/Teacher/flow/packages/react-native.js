//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
