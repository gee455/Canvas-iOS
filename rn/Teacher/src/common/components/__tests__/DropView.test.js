//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import { shallow } from 'enzyme'
import React from 'react'
import { Platform, View } from 'react-native'
import DropView from '../DropView'

jest.mock('Platform', () => ({
  OS: 'ios',
  Version: '11.2',
}))

describe('DropView', () => {
  const warn = console.warn
  beforeEach(() => {
    console.warn = jest.fn()
  })
  afterEach(() => {
    console.warn = warn
  })

  it('renders a native dropview on ios 11 and above', () => {
    // $FlowFixMe
    Platform.Version = '11.2.1'
    const tree = shallow(<DropView><View /></DropView>)
    expect(tree).toMatchSnapshot()
    expect(console.warn).not.toHaveBeenCalled()
  })

  it('warns and renders a normal view on other versions', () => {
    // $FlowFixMe
    Platform.Version = '10.3.1'
    const tree = shallow(<DropView />)
    expect(tree).toMatchSnapshot()
    expect(tree.type()).toBe(View)
    expect(console.warn).toHaveBeenCalledWith('DropView can only be used on iOS 11+')
  })
})
