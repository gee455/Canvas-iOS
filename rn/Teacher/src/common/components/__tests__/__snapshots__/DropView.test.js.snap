// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropView renders a native dropview on ios 11 and above 1`] = `
<_class>
  <View />
</_class>
`;

exports[`DropView warns and renders a normal view on other versions 1`] = `<View />`;
